### PR TITLE
Adds CultureContext coverage

### PR DESCRIPTION
**Pull request by rducom**

### CODE REVIEW

File: NExtends.Tests/CultureContextTest.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            using (new CultureContext(fr))
+            {
+                Assert.Equal(fr, CultureInfo.CurrentCulture);
+                using (new CultureContext(en))
+                {
+                    Assert.Equal(en, CultureInfo.CurrentCulture);
+                }
+                Assert.Equal(fr, CultureInfo.CurrentCulture);
+            }
+        }
```
Comment (nfaugout) [inline NExtends.Tests/CultureContextTest.cs]: Here you just test that CultureInfo is not Equal by reference don't you ?!

So you can thrash this test and add a test that assert that cultureInfo is Equal by value

File: NExtends.Tests/CultureContextTest.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    });
+                }
+                Assert.Equal(fr, CultureInfo.CurrentCulture);
+            }
+        }
+
+        private async Task AsyncDelay(Action action)
+        {
+            await Task.Delay(100);
+            await Task.Yield();
```
Comment (nfaugout) [inline NExtends.Tests/CultureContextTest.cs]: From what I understand of Task.Yeld( ) (https://stackoverflow.com/questions/22645024/when-would-i-use-task-yield), the first await Task.Delay( ) already guarantee that the code of this method will run asynchronously

File: NExtends.Tests/CultureContextTest.cs (hunk 1 above)
Comment (rducom) [reply NExtends.Tests/CultureContextTest.cs]: CultureInfo overrides equals operator : https://referencesource.microsoft.com/#mscorlib/system/globalization/cultureinfo.cs,3c87ed6650eea1ca,references

File: NExtends.Tests/CultureContextTest.cs (hunk 2 above)
Comment (rducom) [reply NExtends.Tests/CultureContextTest.cs]: Task.Yield() is used to trigger SynchronisationContext & ExecutionContext switch : it's the internal process which is used to copy all AsyncLocal<T> values along thread switchs. The purpose of this test is to ensure CultureInfo flows along such thread switchs.

**Pull request merged**